File: hh-back/src/main/java/gloit/hiperionida/helios/model/CompraModel.java
package gloit.hiperionida.helios.model;

import gloit.hiperionida.helios.model.enums.TipoComprobanteEnum;
import gloit.hiperionida.helios.util.model.AbsAuditoriaModel;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@EqualsAndHashCode(callSuper = false)
@Entity
@Getter
@Setter
@SuperBuilder
@Table(name = "compra")
public class CompraModel extends AbsAuditoriaModel {
    private LocalDateTime fecha;
    private Double iva;
    private String numeroComprobante;
    private Double subTotal;
    @Enumerated(EnumType.STRING)
    private TipoComprobanteEnum tipoComprobante;
    @Column(columnDefinition = "TEXT")
    private String notas;

    private Long comprobanteId;
    private Long proveedorId;

    public CompraModel() {
    }

    public CompraModel(Long id, Long creadorId, LocalDateTime creada, Long modificadorId, LocalDateTime modificada, Long eliminadorId, LocalDateTime eliminada, LocalDateTime fecha, Double iva, String numeroComprobante, Double subTotal, TipoComprobanteEnum tipoComprobante, String notas, Long comprobanteId, Long proveedorId) {
        super(id, creadorId, creada, modificadorId, modificada, eliminadorId, eliminada);
        this.fecha = fecha;
        this.iva = iva;
        this.numeroComprobante = numeroComprobante;
        this.subTotal = subTotal;
        this.tipoComprobante = tipoComprobante;
        this.notas = notas;
        this.comprobanteId = comprobanteId;
        this.proveedorId = proveedorId;
    }
}

File: hh-front/src/models/compra_model.js
import { UsuarioModel } from './usuario_model'
import { ArchivoModel } from './archivo_model'
import { ProveedorModel } from './proveedor_model'
import { RemitoModel } from './remito_model'

export class CompraModel {
  constructor () {
    this.id = ''
    this.fecha = ''
    this.iva = ''
    this.numeroComprobante = ''
    this.subTotal = ''
    this.tipoComprobante = ''
    this.notas = ''
    this.comprobante = new ArchivoModel()
    this.pagos = new Array([])
    this.proveedor = new ProveedorModel()
    this.remito = new RemitoModel()

    this.creador = new UsuarioModel()
    this.creada = ''
    this.modificador = new UsuarioModel()
    this.modificada = ''
    this.eliminador = new UsuarioModel()
    this.elimianda = ''
  }
}
